internal/control_test_engine/gnb/ngap/handler.go

```diff
@@ -834,77 +834,80 @@ func HandlerAmfStatusIndication(amf *context.GNBAmf, gnb *context.GNBContext, me
 					if !ok {
 						return true
 					}
-					if amf.GetAmfName() == unavailableGuamiItem.BackupAMFName.Value {
+					if unavailableGuamiItem.BackupAMFName != nil &&
+						amf.GetAmfName() == unavailableGuamiItem.BackupAMFName.Value {
 						backupAmf = amf
 						return false
 					}
 
 					return true
 				})
 
-				amfPool.Range(func(k, v any) bool {
-					oldAmf, ok := v.(*context.GNBAmf)
-					if !ok {
-						return true
-					}
-					for j := 0; j < oldAmf.GetLenPlmns(); j++ {
-						oldAmfSupportMcc, oldAmfSupportMnc := oldAmf.GetPlmnSupport(j)
-
-						if oldAmfSupportMcc == unavailableMcc && oldAmfSupportMnc == unavailableMnc &&
-							reflect.DeepEqual(oldAmf.GetRegionId(), unavailableGuamiItem.GUAMI.AMFRegionID.Value) &&
-							reflect.DeepEqual(oldAmf.GetSetId(), unavailableGuamiItem.GUAMI.AMFSetID.Value) &&
-							reflect.DeepEqual(oldAmf.GetPointer(), unavailableGuamiItem.GUAMI.AMFPointer.Value) {
-
-							log.Info("[GNB][AMF] Remove AMF: [",
-								"Id: ", oldAmf.GetAmfId(),
-								"Name: ", oldAmf.GetAmfName(),
-								"Ipv4: ", oldAmf.GetAmfIp(),
-								"]",
-							)
-
-							tnla := oldAmf.GetTNLA()
-
-							// NGAP UE-TNLA Rebinding
-							uePool := gnb.GetUePool()
-							uePool.Range(func(k, v any) bool {
-								ue, ok := v.(*context.GNBUe)
-								if !ok {
+				if backupAmf != nil {
```

Comment:
Thanks for your PR!
Indentation is becoming a bit too big here, we should do that to avoid adding one layer of indentation:
```go
if backupAmf == nil {
  return
}
amfPool.Range(func(k, v any) bool {
...
```

Reply:
OK. I have fixed it.

BTW, I found that when decoding **AMFStatusIndication** NGAP Message, if `backupAMFName` IE is included in value. The error: `per data out of range` will happened.  I think there's a bug in free5gc/aper and I am going to fix it.